django/contrib/postgres/indexes.py

```diff
@@ -36,6 +36,44 @@ def get_with_params(self):
         return []
 
 
+class BloomIndex(PostgresIndex):
+    suffix = 'bloom'
+
+    def __init__(self, *, length=None, columns=None, **kwargs):
+        if length is not None and not 0 < length <= 4096:
+            raise ValueError(
+                'length must be None or a positive integer up to a maximum of 4096.'
+            )
+        if columns and not (len(columns) <= 32 and all(0 < v <= 4095 for v in columns)):
```

Comment:
Should we check that `columns` is no longer then `fields`?

Reply:
From https://www.postgresql.org/docs/current/bloom.html#id-1.11.7.14.7:

"Parameters for index columns not actually used are ignored."

Having said that, just because PostgreSQL is lax here, it might be nice to let users know?

Reply:
I think we can raise an error in this case.